resource/aws_cognito_identity_pool: openid_connect_provider_arns Showing Ordering Differences

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

Noticed by this consistently failing acceptance test:

```
=== CONT  TestAccAWSCognitoIdentityPool_openidConnectProviderArns
TestAccAWSCognitoIdentityPool_openidConnectProviderArns: resource_aws_cognito_identity_pool_test.go:99: After applying this test step, the plan was not empty.
stdout:
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
~ update in-place
Terraform will perform the following actions:
# aws_cognito_identity_pool.main will be updated in-place
~ resource "aws_cognito_identity_pool" "main" {
allow_unauthenticated_identities = false
arn                              = "arn:aws:cognito-identity:us-west-2:*******:identitypool/us-west-2:29818d19-ab3c-40d5-b3cb-13ec6176ee24"
id                               = "us-west-2:29818d19-ab3c-40d5-b3cb-13ec6176ee24"
identity_pool_name               = "identity pool o9se4jvj2s"
~ openid_connect_provider_arns     = [
- "arn:aws:iam::123456789012:oidc-provider/bar.example.com",
"arn:aws:iam::123456789012:oidc-provider/foo.example.com",
+ "arn:aws:iam::123456789012:oidc-provider/bar.example.com",
]
saml_provider_arns               = []
supported_login_providers        = {}
tags                             = {}
}
Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccAWSCognitoIdentityPool_openidConnectProviderArns (107.41s)
```

The API is returning results in a non-deterministic manner while the resource attribute is declared as an ordered list. As long as the submission ordering is not important, the attribute can be switched to `TypeSet` to fix this difference handling.

### Affected Resource(s)

* aws_cognito_identity_pool

### References

* https://docs.aws.amazon.com/cognitoidentity/latest/APIReference/API_Operations.html

--- a/aws/resource_aws_cognito_identity_pool.go
+++ b/aws/resource_aws_cognito_identity_pool.go
@@ -73,7 +73,7 @@ func resourceAwsCognitoIdentityPool() *schema.Resource {
 			},
 
 			"openid_connect_provider_arns": {
-				Type:     schema.TypeList,
+				Type:     schema.TypeSet,
 				Optional: true,
 				Elem: &schema.Schema{
 					Type:         schema.TypeString,
@@ -130,7 +130,7 @@ func resourceAwsCognitoIdentityPoolCreate(d *schema.ResourceData, meta interface
 	}
 
 	if v, ok := d.GetOk("openid_connect_provider_arns"); ok {
-		params.OpenIdConnectProviderARNs = expandStringList(v.([]interface{}))
+		params.OpenIdConnectProviderARNs = expandStringSet(v.(*schema.Set))
 	}
 
 	if v, ok := d.GetOk("tags"); ok {
@@ -221,7 +221,7 @@ func resourceAwsCognitoIdentityPoolUpdate(d *schema.ResourceData, meta interface
 	}
 
 	if v, ok := d.GetOk("openid_connect_provider_arns"); ok {
-		params.OpenIdConnectProviderARNs = expandStringList(v.([]interface{}))
+		params.OpenIdConnectProviderARNs = expandStringSet(v.(*schema.Set))
 	}
 
 	if v, ok := d.GetOk("saml_provider_arns"); ok {

--- a/website/docs/r/cognito_identity_pool.markdown
+++ b/website/docs/r/cognito_identity_pool.markdown
@@ -53,7 +53,7 @@ The Cognito Identity Pool argument layout is a structure composed of several sub
 * `developer_provider_name` (Optional) - The "domain" by which Cognito will refer to your users. This name acts as a placeholder that allows your
 backend and the Cognito service to communicate about the developer provider.
 * `cognito_identity_providers` (Optional) - An array of [Amazon Cognito Identity user pools](#cognito-identity-providers) and their client IDs.
-* `openid_connect_provider_arns` (Optional) - A list of OpendID Connect provider ARNs.
+* `openid_connect_provider_arns` (Optional) - Set of OpendID Connect provider ARNs.
 * `saml_provider_arns` (Optional) - An array of Amazon Resource Names (ARNs) of the SAML provider for your identity.
 * `supported_login_providers` (Optional) - Key-Value pairs mapping provider names to provider app IDs.
 * `tags` - (Optional) A map of tags to assign to the Identity Pool.

